src/components/Calculator.js

```diff
@@ -1,30 +1,57 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 export default class Calculator extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      calcInitial: 0,
+    };
+    this.calc = this.calc.bind(this);
+  }
+
+  calc(e) {
+    const value = e.target.innerText;
+    const { calcInitial } = this.state;
+    const calcResult = calculate(calcInitial, value);
+    this.setState({ calcInitial: calcResult });
+    const display = document.querySelector('.calcFirtsRow');
+
+    if (calcResult.total === null && calcResult.next !== null) {
+      display.textContent = calcResult.next;
+    } else if (calcResult.total !== null && calcResult.next !== null) {
+      display.textContent = calcResult.next;
+    } else if (calcResult.total !== null && calcResult.next == null) {
+      display.textContent = calcResult.total;
+    } else {
+      display.textContent = 0;
+    }
+  }
+
   render() {
     return (
       <div className="calcBody">
         <div className="calcFirtsRow">
           <p>0</p>
         </div>
-        <button type="button" className="leftCalc one">1</button>
-        <button type="button" className="leftCalc two">2</button>
-        <button type="button" className="leftCalc three">3</button>
-        <button type="button" className="leftCalc four">4</button>
-        <button type="button" className="leftCalc five">5</button>
-        <button type="button" className="leftCalc six">6</button>
-        <button type="button" className="leftCalc seven">7</button>
-        <button type="button" className="leftCalc eight">8</button>
-        <button type="button" className="leftCalc nine">9</button>
-        <button type="button" className="leftCalc cero">0</button>
-        <button type="button" className="leftCalc AC">A/C</button>
-        <button type="button" className="leftCalc plusMinus">+/-</button>
-        <button type="button" className="leftCalc reminder">%</button>
-        <button type="button" className="rightCalc div">&divide;</button>
-        <button type="button" className="rightCalc mult">X</button>
-        <button type="button" className="rightCalc sub">-</button>
-        <button type="button" className="rightCalc plus">+</button>
-        <button type="button" className="rightCalc equal">=</button>
+        <button type="button" className="leftCalc one" onClick={this.calc}>1</button>
+        <button type="button" className="leftCalc two" onClick={this.calc}>2</button>
```

Comment:
Please add the dot (`.` ) button that allows users to handle decimals.

![image](https://user-images.githubusercontent.com/38088818/170534376-c613fe85-3a25-4a44-bd94-950aa3c7b2ae.png)

Reply:
@KDlamini  Oh I miss that, Thanks I will make the change